components/omega/doc/design/VertCoord.md

```diff
@@ -0,0 +1,294 @@
+(omega-design-vert-coord)=
+# Vertical Coordinate
+
+## 1 Overview
+The vertical coordinate module will be responsible for computing and storing information related to the vertical mesh in Omega.
+Since Omega will be a non-Boussinesq model, the vertical independent variable will be pressure- as opposed to $z$-based.
+Similar to MPAS-Ocean's support for tilted height ($z^\star$) coordinates, Omega V1 will support a general ($p^\star$) vertical coordinate, where mass thickness $h$ varies in proportion to the total mass thickness of the water column. 
+The prognostic variable in the layered continuity equation is mass thickness, which can be used to calculate the total pressure at a given vertical level.
+The vertical height of a given level is still necessary in the model to compute sea level, the geopotential, and derivatives with respect to $z$ (vertical mixing, etc).
+The height of a layer ($\Delta z$) can be found using the mass thickness, density and known bottom elevation relative to the geoid (positive up).
+The vertical coordinate module will also serve as a container for information related to the variable number of vertical levels for a given ocean column due to variations in bottom elevation and ice shelf cavities.
```

Comment:
I know I was on leave, so maybe I missed a decision somewhere, but technically its the active number of layers that is variable, correct? There will still be the same number of vertical levels for every column, just the number of layers we loop over (or mask) will change based upon bottom elevation and ice shelf cavities? 

Maybe this is nit-picky, but maybe something like `variable number of active vertical levels` or whatever terminology we want to use to differentiate between `NVertLevels` and the quantity that is `maxLevelCell - minLevelCell`.